databases/metadata/src/main/functions/04_ontology_hierarchy.sql

```diff
@@ -0,0 +1,32 @@
+SET search_path = public, pg_catalog;
+
+--
+-- A function that returns all of the ontology subclasses of the class with the given
+-- ontology_version and IRI.
+--
+CREATE OR REPLACE FUNCTION ontology_class_hierarchy(VARCHAR, VARCHAR)
```

Comment:
Sure, here's an `EXPLAIN ANALYZE` for an `ontology_class_hierarchy` query using the Operation class as a root:
```sql
metadata=> EXPLAIN ANALYZE SELECT * FROM ontology_class_hierarchy('http://edamontology.org/1.14/2016/04/20/20-39-12.5', 'http://edamontology.org/operation_0004');
                                                           QUERY PLAN                                                           
--------------------------------------------------------------------------------------------------------------------------------
 Function Scan on ontology_class_hierarchy  (cost=0.25..10.25 rows=1000 width=96) (actual time=86.554..87.403 rows=738 loops=1)
 Planning time: 0.319 ms
 Execution time: 88.533 ms
(3 rows)
```

This [EDAM Ontology Operation class](http://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Foperation_0004) has the larger hierarchy of the 2 that the admins plan to use for the new app category hierarchies.

This query was run against our dev db that had 15 versions of the same ontology hierarchies loaded, for a total of 13599 rows in the `ontology_hierarchies` table. This many ontologies and hierarchies are not likely to be loaded in production any time soon.